networkmanager/networkmanager.go

```diff
@@ -644,7 +655,7 @@ func (manager *NetworkManager) postNetworkClear(networkID string) error {
 	return nil
 }
 
-func (manager *NetworkManager) prepareRuntimeConfig(instanceID, networkID, hostname string, aliases []string) (
+func (manager *NetworkManager) prepareRuntimeConfig(instanceID, networkID string, params *NetworkParams) (
```

Comment:
You should not modify param.Aliases here. It makes code unclear. Add new function which creates hosts list based on network parameters (hostname, aliases, ids, etc.) and pass this list to this function.